src/JsonSchema/Constraints/Constraint.php

```diff
@@ -72,33 +47,13 @@ public function getUriRetriever()
     public function getFactory()
     {
         if (!$this->factory) {
-            $this->factory = new Factory($this->getSchemaStorage(), $this->getUriRetriever(), $this->checkMode);
+            $this->factory = new Factory();
```

Comment:
If the factory is now always created in the constructor, there is no point in doing `if (!$this->factory)` in the public `getFactory` method anymore.